
// Code generated by controller-gen. DO NOT EDIT.
// <copyright file="ErrorEvent.go" company="Microsoft Corporation">
// Copyright (c) Microsoft Corporation. All rights reserved.
// </copyright>
// <autogenerated />

package discoveryextensionevents

import (
	"fmt"
)

func GetUnhandledException(errorCode string, errorMessage string, errortype string) ErrorEvent {
    message := fmt.Sprintf("Internal error occured: %s, Message: %s, %s,.", errorCode, errorMessage, errortype)
    possibleCausess := "The operation failed due to an internal error."
    recommendedAction := "Retry the operation. If the issue persists, contact support."
    return getErrorEvent(
        UnhandledException,
        message,
        possibleCausess,
        recommendedAction,
        map[string]string{
                    "ErrorCode": errorCode ,
                    "ErrorMessage": errorMessage ,
                    "Errortype": errortype ,
                });
        }
func GetNullRefException(reason string) ErrorEvent {
    message := fmt.Sprintf("Internal error occured: Message: %s,.", reason)
    possibleCausess := "The operation failed due to an internal error."
    recommendedAction := "Retry the operation. If the issue persists, contact support."
    return getErrorEvent(
        NullRefException,
        message,
        possibleCausess,
        recommendedAction,
        map[string]string{
                    "Reason": reason ,
                });
        }
